lib/matplotlib/patches.py

```diff
@@ -1209,21 +1209,15 @@ def _recompute_path(self):
             # followed by a reversed and scaled inner ring
             v1 = arc.vertices
             v2 = arc.vertices[::-1] * (self.r - self.width) / self.r
-            v = np.concatenate([v1, v2, [v1[0, :], (0, 0)]])
-            c = np.concatenate([
-                arc.codes, arc.codes, [connector, Path.CLOSEPOLY]])
-            c[len(arc.codes)] = connector
+            v = np.concatenate([v1, v2, [(0, 0)]])
+            c = [*arc.codes, connector, *arc.codes[1:], Path.CLOSEPOLY]
```

Comment:
In the case of a full anuulus will this result in an extra line connecting the inner and outer rings?

Reply:
No, because for full annuli(?) connector is MOVETO, which defines a new initial point for a subpath, and CLOSEPOLY closes to that initial point.  You can check with e.g.
```
from pylab import *; w = mpl.patches.Wedge((.5, .5), .3, 0, 360, width=.1, linewidth=5, facecolor="none", edgecolor="k"); figure().add_subplot(aspect=1).add_artist(w)
```

Reply:
Shouldn't the outer arc be `CLOSEPOLY`'d as well, then? I agree it won't break, but that seems the semantically correct thing to do (and the existing code didn't do it correctly.)

Reply:
Fair, but I think that's something that should be fixed at the level of Path.arc(), not here.